### ISSUE TITLE
Deprecated TimeoutQueue so we can remove it

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2536]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2536|
|Type|task|
|Created|2007-03-23 12:02:13Z|

This code isn't used anywhere in Twisted and it doesn't do anything worth using anyway.

<details><summary>Searchable metadata</summary>

```
trac-id__2536 2536
type__task task
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1174651333000000 1174651333000000
changetime__1181410763000000 1181410763000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

and IIRC Python provides its functionality these days anyway.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve removed owner|
|-|-|

Ready to review in timeoutqueue-2536. I've created [#2626](https://github.com/twisted/twisted/issues/2626) to have the functionality for testing DeprecationWarning. We can also wait for this one, it may be cleaner.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @therve|
|-|-|

Let's delay that until [#2626](https://github.com/twisted/twisted/issues/2626) is merged.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve removed owner|
|-|-|

Ready to review in timeoutqueue-2536-2.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @therve|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

(In [20510]) Merge timeoutqueue-2536-2

Author: therve
Reviewer: exarkun
Fixes [#2536](https://github.com/twisted/twisted/issues/2536)

Deprecate twisted.python.timeoutqueue, as it's unused in Twisted and stdlib
Queue module provides same functionalities.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
